### PR TITLE
removed the fromUID of messages that go through rust

### PR DESCRIPTION
**Pull request by xkevin190**

### CODE REVIEW

File: __tests__/Views/chats/chatForm-test.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import React from 'react';
+import { shallow } from 'enzyme';
+import ChatForm from '../../../src/views/home/ChatForm';
+import mock from '../../../__Mocks__/dataMock';
+
+const sendChatMock = jest.fn();
+const mockOpenModal = jest.fn();
+const sendMessageWithSongMock = jest.fn();
+
+
```
Comment (luisan00) [inline __tests__/Views/chats/chatForm-test.js:11]: Extra blank line
```suggestion
```

File: __tests__/Views/chats/chatForm-test.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      sendMessagesWithSound={sendMessageWithSongMock}
+      screenProps={screenProps}
+    />
+  );
+
+  test('render all component', () => {
+    expect(wrapper.instance()).toBeDefined();
+  });
+
+
```
Comment (luisan00) [inline __tests__/Views/chats/chatForm-test.js:41]: Extra line 
```suggestion
```

File: __tests__/Views/chats/chatForm-test.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  test('componentDidMount', async () => {
+    const instance = wrapper.instance();
+    await instance.componentDidMount();
+    expect(instance.state.hasPermission).toBe(true);
+  });
+
+
```
Comment (luisan00) [inline __tests__/Views/chats/chatForm-test.js:48]: Extra line
```suggestion
```

File: __tests__/Views/chats/message-test.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  name: undefined,
+  shippingTime: 1602013395803,
+  status: 'pending',
+  time: 1602013395803,
+  timestamp: 1602013395786,
+  toUID: 'broadcast',
+  type: 1
+}];
+
+
```
Comment (luisan00) [inline __tests__/Views/chats/message-test.js:21]: Extra line 
```suggestion
```

File: __tests__/Views/chats/message-test.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+const mockContact = [{
+  hashUID: '1ed4091506f77825bccb4dadcf89f1b62ff3257d2afc3886bf859c0330120a27',
+  name: 'Luis',
+  nodeAddress: '/ip4/85.53.137.144/tcp/4444',
+  picture: null,
+  uid: '16Uiu2HAm2BXA1TTUE7vvSu3U4YURxb4FwTYjfrwnZN618jrYBpNS'
+}];
+
+
```
Comment (luisan00) [inline __tests__/Views/chats/message-test.js:34]:
```suggestion
```

File: __tests__/Views/chats/message-test.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      onSelected={onLongPressMock}
+
+    />
+  );
+
+  test('render all component', () => {
+    expect(wrapper.instance()).toBeDefined();
+  });
+
+
```
Comment (luisan00) [inline __tests__/Views/chats/message-test.js:54]:
```suggestion
```

File: __tests__/Views/chats/message-test.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
+import '../../../__Mocks__';
+import React from 'react';
+import { shallow } from 'enzyme';
+import { ReceiveMessage } from '../../../src/views/home/Messages';
+
+
```
Comment (luisan00) [inline __tests__/Views/chats/message-test.js:6]:
```suggestion
```

File: android/app/src/main/java/DeviceInfo/Utils.java [hunk 8]
```diff
@@ -178,9 +181,14 @@ public static String getOurVersion(Context ctx) {
...
+        JSONObject obj = new JSONObject(json);
+        obj.put("fromUID", peerID);
 
-    public static int suma(int a, int b){
-        return a + b;
+        return  obj.toString();
     }
 
+
+
```
Comment (luisan00) [inline android/app/src/main/java/DeviceInfo/Utils.java:193]:
```suggestion
```

File: src/views/home/ChatForm.js [hunk 9]
```diff
@@ -156,8 +156,10 @@ export default class ChatForm extends Component {
   send = async () => {
     const { user, navigation, setChat } = this.props;
     const toUID = navigation.params ? navigation.params.uid : 'broadcast';
+
+    console.log("dios123", user);
```
Comment (luisan00) [inline src/views/home/ChatForm.js:160]: 😂

File: android/app/src/main/java/io/locha/p2p/service/EventsDispatcher.java [hunk 10]
```diff
@@ -58,11 +59,15 @@ public  void setApplicationContext(ReactApplicationContext context) {
...
-        map.putString("peerId", peerId);
-        map.putString("contents", contents);
-        sendEvent(this.reactContext, "newMessage", map);
+        try {
+            Log.d(TAG, "newMessage");
+
+            String stringify = Utils.parseJSON(contents, peerId);
+            sendEvent(this.reactContext, "newMessage", stringify);
+        } catch (Exception err) {
+            Log.e(TAG, " something failed to the parse the json:", err );
```
Comment (luisan00) [inline android/app/src/main/java/io/locha/p2p/service/EventsDispatcher.java:68]:
```suggestion
            Log.e(TAG, " something failed trying to parse the JSON:", err );
```

**Review by luisan00 — CHANGES_REQUESTED** [review]

**Review by luisan00 — APPROVED** [review]

File: android/app/src/main/java/DeviceInfo/Utils.java [hunk 11]
```diff
@@ -178,9 +180,11 @@ public static String getOurVersion(Context ctx) {
...
 
 
+    public static String parseJSON(String json, String peerID) throws JSONException {
+        
+        JSONObject obj = new JSONObject(json);
+        obj.put("fromUID", peerID);
 
-    public static int suma(int a, int b){
-        return a + b;
+        return  obj.toString();
```
Comment (jeandudey) [inline android/app/src/main/java/DeviceInfo/Utils.java:188]: This is small enough to put it on `onNewMessage` :wink:

File: android/app/src/main/java/io/locha/p2p/service/EventsDispatcher.java [hunk 12]
```diff
@@ -58,11 +59,15 @@ public  void setApplicationContext(ReactApplicationContext context) {
...
-        map.putString("peerId", peerId);
-        map.putString("contents", contents);
-        sendEvent(this.reactContext, "newMessage", map);
+        try {
+            Log.d(TAG, "newMessage");
+
+            String stringify = Utils.parseJSON(contents, peerId);
+            sendEvent(this.reactContext, "newMessage", stringify);
+        } catch (Exception err) {
+            Log.e(TAG, " something failed trying parse the JSON:", err );
```
Comment (jeandudey) [inline android/app/src/main/java/io/locha/p2p/service/EventsDispatcher.java:68]:
```suggestion
            Log.e(TAG, "something failed trying parse the message JSON:", err );
```

**Review by jeandudey — CHANGES_REQUESTED** [review]
Only the above comment, everuthing else is fine

File: android/app/src/main/java/DeviceInfo/Utils.java (hunk 11 above)
Comment (xkevin190) [reply android/app/src/main/java/DeviceInfo/Utils.java:188]: but,This little piece of code has nothing to do with the event, that's why it is separate hehe

**Review by jeandudey — APPROVED** [review]

**Pull request merged**